research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,434 @@
+
+**Example:** Invoking the `button` element in this example should show/hide the `popup` element,
+depending on whether or not the `popup` element is currently shown:
+
+```html
+<button id="menuButton" popup="menuPopup">Menu</button>
+<popup id="menuPopup" role="menu" anchor="menuButton">
+  <!-- Markup for menuitems goes here -->
+</popup>
+```
+
+The `popup` attribute is supported on a subset of
+[interactive elements](https://html.spec.whatwg.org/multipage/interactive-elements.html):
+
+- `button`
+- `input` in the `button` state
+- `input` in the `email`, `number`, `search`, `tel`, `text`, or `url` states
+
+#### Showing/hiding a `popup` element via the `popup` attribute
+
+When an element with the `popup` attribute specified receives a user interaction that should show
+the related `popup` element, this element shall be known as the `popup` element's _invoker_.
+
+<p class="note">
+  Multiple elements may have a popup attribute that refers to the same popup element. A given popup
+  element only has 0 or 1 invokers at a given time. For example, if the popup element was shown
+  programmatically, the invoker is null. Otherwise, the element that the user actually interacted
+  with to show the popup is its sole invoker.
+</p>
+
+If the _invoker_ is a `button` element or an `input` element in the `button` state, and the
+_invoker_ is activated; OR the _invoker_ is an input in the `email`, `number`, `search`, `tel`,
+`text`, or `url` states and focus is set to the _invoker_, the user agent must run the following
+steps:
+
+1. Let _subject_ be the first element in tree order with the ID referenced by the `popup` attribute
+   specified on the _invoker_. If there is no such element, or the _subject_ is not a `popup`
+   element, then return.
+2. Get the _subject_'s _open state_.
+3. If _subject_'s _open state_ is `true`, run
+   [hiding a `popup` element steps](#hiding-current-popups).
+4. Otherwise, run [showing a `popup` element steps](#showing-popup-steps), with _candidate subject_
+   set to _subject_ and _invoker_ set to the current element _invoker_.
+
+<p class="question">
+  Should we have a way to suppress these steps upon focus, so that the author can enable the popup
+  to be shown upon text entry (or some other logic) instead?
+</p>
+
+<h3 id="showing-popup-steps">
+  Showing a <code>popup</code> element steps
+</h3>
+
+The **showing a `popup` element steps**, given a popup element _candidate subject_ and an optional
+element _invoker_, and an optional element _anchor element_, are:
+
+1. Run the [hiding currently-shown `popup` elements steps](#hiding-current-popups), with the
+   _candidate subject_, optional element _invoker_, and optional element _anchor element_.
+2. Add the _candidate subject_ to _candidate subject_'s
+   [node document](https://dom.spec.whatwg.org/#concept-node-document)'s [popup stack](#popup-stack).
+3. Set the `open` IDL attribute on the _candidate subject_ to `true`.
```

Comment:
I added step 4 here based on the previous feedback to hold this in an internal state mechanism, so satisfying this comment would be a case of removing (3).

What is a bit unclear to me is why we wouldn't want to have the IDL attr reflect the UA-known state of the popup. Couldn't that create clashes in the other direction, because the state is out of sync/inaccurate?

Reply:
I think the IDL attribute should reflect the internal state. I expect its spec to be "The open getter steps are to return this's open state" (or whatever you named the internal state; I haven't looked at the diff yet :)).

Reply:
Ah, I better understand the feedback now. Thanks!

Reply:
Addressed in 5fc5041409d28c395e81cf30f039a7c0e269b7fe